tee-worker/omni-executor/aa-contracts/src/core/BasePaymaster.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+/* solhint-disable reason-string */
+
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "../interfaces/IPaymaster.sol";
+import "../interfaces/IEntryPoint.sol";
+import "./UserOperationLib.sol";
+/**
+ * Helper class for creating a paymaster.
+ * provides helper methods for staking.
+ * Validates that the postOp is called only by the entryPoint.
+ */
+
+abstract contract BasePaymaster is IPaymaster, Ownable2Step {
+    IEntryPoint public immutable entryPoint;
+
+    uint256 internal constant PAYMASTER_VALIDATION_GAS_OFFSET = UserOperationLib.PAYMASTER_VALIDATION_GAS_OFFSET;
+    uint256 internal constant PAYMASTER_POSTOP_GAS_OFFSET = UserOperationLib.PAYMASTER_POSTOP_GAS_OFFSET;
+    uint256 internal constant PAYMASTER_DATA_OFFSET = UserOperationLib.PAYMASTER_DATA_OFFSET;
+
+    constructor(IEntryPoint _entryPoint) Ownable(msg.sender) {
+        _validateEntryPointInterface(_entryPoint);
+        entryPoint = _entryPoint;
+    }
+
+    // Sanity check: make sure this EntryPoint was compiled against the same
+    // IEntryPoint of this paymaster
+    function _validateEntryPointInterface(IEntryPoint _entryPoint) internal virtual {
+        require(
+            IERC165(address(_entryPoint)).supportsInterface(type(IEntryPoint).interfaceId),
+            "IEntryPoint interface mismatch"
+        );
+    }
+
+    /// @inheritdoc IPaymaster
+    function validatePaymasterUserOp(PackedUserOperation calldata userOp, bytes32 userOpHash, uint256 maxCost)
```

Comment:
Do we need to add a pause function, so that we can pause this part at any time in case there are issues?  
```
  function validatePaymasterUserOp(...)
        external
        override
        whenNotPaused 
        returns (bytes memory context, uint256 validationData)
    {
        // ...
    }
    
    function emergencyPause() external onlyOwner {
        _pause();
    }
    function unpause() external onlyOwner {
        _unpause();
    }

Reply:
added. 

It could be paused by removing all `authorizedBundlers` from `SimplePaymaster` but I've added pausable feature in `BasePaymaster`